node/core/backing/src/lib.rs

```diff
@@ -0,0 +1,1454 @@
+enum FromJob {
+	RuntimeApiMessage(RuntimeApiMessage),
+	CandidateValidation(CandidateValidationMessage),
+	CandidateSelection(CandidateSelectionMessage),
+	Provisioner(ProvisionerMessage),
+	PoVDistribution(PoVDistributionMessage),
+	StatementDistribution(StatementDistributionMessage),
+}
+
+impl From<FromJob> for AllMessages {
+	fn from(f: FromJob) -> Self {
+		match f {
+			FromJob::RuntimeApiMessage(msg) => AllMessages::RuntimeApi(msg),
+			FromJob::CandidateValidation(msg) => AllMessages::CandidateValidation(msg),
+			FromJob::CandidateSelection(msg) => AllMessages::CandidateSelection(msg),
+			FromJob::StatementDistribution(msg) => AllMessages::StatementDistribution(msg),
+			FromJob::PoVDistribution(msg) => AllMessages::PoVDistribution(msg),
+			FromJob::Provisioner(msg) => AllMessages::Provisioner(msg),
+		}
+	}
+}
+
+// It looks like it's not possible to do an `impl From` given the current state of
+// the code. So this does the necessary conversion.
+fn primitive_statement_to_table(s: &SignedFullStatement) -> TableSignedStatement {
+	let statement = match s.payload() {
+		Statement::Seconded(c) => TableStatement::Candidate(c.clone()),
+		Statement::Valid(h) => TableStatement::Valid(h.clone()),
+		Statement::Invalid(h) => TableStatement::Invalid(h.clone()),
+	};
+
+	TableSignedStatement {
+		statement,
+		signature: s.signature().clone(),
+		sender: s.validator_index(),
+	}
+}
+
+// finds the first key we are capable of signing with out of the given set of validators,
+// if any.
+fn signing_key(validators: &[ValidatorId], keystore: &KeyStorePtr) -> Option<ValidatorPair> {
+	let keystore = keystore.read();
+	validators.iter()
+		.find_map(|v| {
+			keystore.key_pair::<ValidatorPair>(&v).ok()
+		})
+		.map(|pair| pair)
+}
+
+impl CandidateBackingJob {
+	/// Run asynchronously.
+	async fn run(mut self) -> Result<(), Error> {
+		while let Some(msg) = self.rx_to.next().await {
+			match msg {
+				ToJob::CandidateBacking(msg) => {
+					self.process_msg(msg).await?;
+				}
+				_ => break,
+			}
+		}
+
+		Ok(())
+	}
+
+	async fn issue_candidate_invalid_message(
+		&mut self,
+		candidate: AbridgedCandidateReceipt,
+	) -> Result<(), Error> {
+		self.tx_from.send(FromJob::CandidateSelection(
+				CandidateSelectionMessage::Invalid(self.parent, candidate)
+		)).await?;
+
+		Ok(())
+	}
+
+	/// Validate the candidate that is requested to be `Second`ed and distribute validation result.
+	async fn validate_and_second(
+		&mut self,
+		candidate: AbridgedCandidateReceipt,
+		pov: PoVBlock,
+	) -> Result<ValidationResult, Error> {
+		let valid = self.request_candidate_validation(candidate.clone(), pov).await?;
+		let statement = match valid {
+			ValidationResult::Valid => {
+				// make PoV available for later distribution. Send data to the availability
+				// store to keep. Sign and dispatch `valid` statement to network if we
+				// have not seconded the given candidate.
+				Statement::Seconded(candidate)
```

Comment:
The comment says something about making the PoV available but the code does not do it here.

Reply:
We should probably update `issued_validity` here as well